### ISSUE TITLE
GuiScrollBar split change causes cursor jump

### ISSUE THREAD

**Issue by jgehrig** [equalsraf/neovim-qt#661]
**Repro Steps:**
1. `nvim-qt -- -u NONE`
2. `:put=range(0,100)`
3. `gg`
4. `:vert h $HOME`
5. Switch back to original buffer: `<C-w>` `l`
6. Notice the cursor position jumps

![cursor-jump-guiscrollbar](https://user-images.githubusercontent.com/11207308/77711195-b038fb00-6fa6-11ea-96e3-9b50f3e5ce73.gif)

**Breaking Commit:** 1c19068915f3f02a3a6abd4f044c0502b1b6648d

**Comment by jgehrig** [comment]
Issue #668 Fixed this issue for `:GuiScrollBar 0`.

Pull Request #677 fixes this issue when `win_viewport` is supported.

A fix is still required for when this feature is enabled, but `win_viewport` is not supported (`nvim <0.5.0`). As a worst-case fix, the scrollbar can be set read-only for older clients.

**Comment by jgehrig** [comment]
Doh! :facepalm:

There was a subtly incorrect order of operations issue. The call to `setMaximum` can result in a `valueChanged` signal if the buffer size decreases. This function should be called within the signal blocker

```C++
// Block valueChanged signals for scroll events triggered by Neovim.
{
	QSignalBlocker blockValueChanged{ this };

	setMaximum(bufferSize);
	setPageStep(windowHeight);
	setSliderPosition(minLine);
}
```

@Shatur95

A fix will be included in Pull Request #677.

**Comment by Shatur95** [comment]
Thank you!
